core/spi/src/main/java/org/jboss/arquillian/core/spi/Manager.java

```diff
@@ -46,4 +46,10 @@
 
     // clean
     void shutdown();
+
+    // adds extension specified via annotation @Observer
+    void addExtension(Class<?> extension) throws Exception;
```

Comment:
What confuses me is that the method is called `addExtension` but we are not registering an extension at all, just an `Observer`. I mean I expect that a method called `addEXtension` I'd add a class of type `LoadableExtension`.

Reply:
This makes me think that maybe we could either extend this to not only set `@Observe` but also `@Extension` which could register a new `LoadableExtension`. This means that with some refactoring on extensions side, we could enable extensions just for one test and not the whole suite. Initially this would be an improvement for example for extensions like Cube where you can mix in same project unit tests and IT tests.

Reply:
The thing is that I add an [Extension](https://github.com/arquillian/arquillian-core/blob/master/core/spi/src/main/java/org/jboss/arquillian/core/spi/Extension.java) ([ExtensionImpl](https://github.com/arquillian/arquillian-core/blob/master/core/impl-base/src/main/java/org/jboss/arquillian/core/impl/ExtensionImpl.java) in particular). See also this line: https://github.com/arquillian/arquillian-core/pull/161/files#diff-ed569ea227c8e0ed8e345149cccd827fR325

The `LoadableExtension` and `Extension` are two different things in the sense of the implementation of Arquillian Core. Also, when we print information about loaded observers, interceptors, etc. [we express it as an extension](https://github.com/arquillian/arquillian-core/blob/master/core/impl-base/src/main/java/org/jboss/arquillian/core/impl/RuntimeLogger.java#L31-L35).

As this is related to the inner implementation, then I would keep the name as it is. Otherwise, we would need to change the naming across the whole project.